### ISSUE TITLE
Missing state for ImageStatus.

### ISSUE THREAD

**Issue by markhibberd** [brendanhay/amazonka#116]
```
Serialization error. Abbreviation: EC2, Error: Failed reading: Failure parsing ImageState from "pending"
```

Pending isn't in the generated code - 
https://github.com/brendanhay/amazonka/blob/develop/amazonka-ec2/gen/Network/AWS/EC2/Types.hs#L10163-L10168 - but I haven't started digging as to see why yet.

**Comment by brendanhay** [comment]
The reason is due to EC2 updates WRT service definitions, botocore (which amazonka tracks) now [has the correct](https://github.com/boto/botocore/blob/80b0161316dbbe355149f619da4ecf1565123e93/botocore/data/ec2/2015-04-15/service-2.json#L7709) enumeration values, where [amazonka does not](https://github.com/brendanhay/amazonka/blob/279e178841a9d7b7f0be62dacd07a2537a0e32ea/gen/input/ec2/2014-10-01.normal.json#L6450). It's a case of updating the definitions and regenerating the code - both of which I'll take care of.

Thanks for bringing it to my attention.

**Comment by charleso** [comment]
@brendanhay I know you'll get around to this, but does this look right?

https://github.com/ambiata/amazonka/pull/2

**Comment by brendanhay** [comment]
It looks correct from a cursory glance, @charleso.

I'm in the process of updating the entire suite of libraries from the current botocore definitions, which requires some modifications to the upstream naming of definitions due to changes in format. (There is a branch that takes care of this, but it's not ready to be merged.)

I'll release the changes to Hackage if the builds and CI processes complete successfully, and let you know.

**Comment by charleso** [comment]
Thanks @brendanhay. Just out of curiosity/laziness what changed upstream?

**Comment by brendanhay** [comment]
They are moving to a [different structure](https://github.com/aws/aws-sdk-ruby/tree/be9b1311fb23eb376f78811325ce5d6a6f2e3187/aws-sdk-core/apis/autoscaling/2011-01-01) where there are some file/naming layout differences, and the documentation will potentially be separated out from the actual shape definitions into another file. 

The files are also now versioned with a suffix.

**Comment by brendanhay** [comment]
The regenerated types have been released and are available as [version 0.3.5 on Hackage.](hackage.haskell.org/package/amazonka-0.3.5)

**Comment by brendanhay** [comment]
I'll close the issue, if you have an problems feel free to reopen or create a new issue.
